pyoptsparse/pyOpt_history.py

```diff
@@ -257,10 +257,8 @@ def getConNames(self):
         # only do this if we open the file with 'r' flag
         if self.flag != "r":
             return
-        conNames = list(self.conInfo.keys())
-        for con in conNames:
-            if self.optProb.constraints[con].linear:
-                conNames.remove(con)
+        # we remove linear constraints
+        conNames = [con for con in self.conInfo.keys() if not self.optProb.constraints[con].linear]
```

Comment:
This is much cleaner! Just to be sure, this loop generates an identical object to `list()`, right?